indexer/services/comlink/src/controllers/api/v4/perpetual-positions-controller.ts

```diff
@@ -150,6 +156,143 @@ class PerpetualPositionsController extends Controller {
       }),
     };
   }
+
+  @Get('/parentSubaccountNumber')
+  async listPositionsForParentSubaccount(
+    @Query() address: string,
+      @Query() parentSubaccountNumber: number,
+      @Query() status?: PerpetualPositionStatus[],
+      @Query() limit?: number,
+      @Query() createdBeforeOrAtHeight?: number,
+      @Query() createdBeforeOrAt?: IsoString,
+  ): Promise<PerpetualPositionResponse> {
+    // Get subaccountIds for all child subaccounts of the parent subaccount
+    // Create a record of subaccountId to subaccount number
+    const childIdtoSubaccountNumber: Record<string, number> = {};
+    getChildSubaccountNums(parentSubaccountNumber).forEach(
+      (subaccountNum: number) => {
+        childIdtoSubaccountNumber[SubaccountTable.uuid(address, subaccountNum)] = subaccountNum;
+      },
+    );
+    const childSubaccountIds: string[] = Object.keys(childIdtoSubaccountNumber);
+
+    const [
+      positions,
+      markets,
+    ]: [
+      PerpetualPositionFromDatabase[],
+      MarketFromDatabase[],
+    ] = await Promise.all([
+      PerpetualPositionTable.findAll(
+        {
+          subaccountId: childSubaccountIds,
+          status,
+          limit,
+          createdBeforeOrAtHeight: createdBeforeOrAtHeight
+            ? createdBeforeOrAtHeight.toString()
+            : undefined,
+          createdBeforeOrAt,
+        },
+        [QueryableField.LIMIT],
+      ),
+      MarketTable.findAll(
+        {},
+        [],
+      ),
+    ]);
+
+    const openPositionsExist: boolean = positions.some(
+      (position: PerpetualPositionFromDatabase) => position.status === PerpetualPositionStatus.OPEN,
+    );
+
+    let updatedPerpetualPositions:
+    PerpetualPositionWithFunding[] = initializePerpetualPositionsWithFunding(positions);
+
+    // Only update the funding for open positions
+    if (openPositionsExist) {
+      const subaccountIds: string[] = updatedPerpetualPositions.map(
+        (position: PerpetualPositionFromDatabase) => position.subaccountId,
+      );
+      const [
+        subaccounts,
+        latestBlock,
+      ]: [
+        SubaccountFromDatabase[],
+        BlockFromDatabase | undefined,
+      ] = await Promise.all([
+        SubaccountTable.findAll(
+          {
+            id: subaccountIds,
+          },
+          [],
+        ),
+        BlockTable.getLatest(),
+      ]);
+
+      if (subaccounts.length === 0 || latestBlock === undefined) {
+        throw new NotFoundError(
+          `Found OPEN perpetual positions but no subaccounts with address ${address} ` +
+            `and parent Subaccount number ${parentSubaccountNumber}`,
+        );
+      }
```

Comment:
I don't think this is possible, because we have a foreign key on the positions table for the subaccount to exist

Reply:
I agree it shouldn't be possible. Also brought over from the original API. Should I remove it from there too?